modules/n1ql/pages/n1ql-language-reference/using-ai.adoc

```diff
@@ -133,67 +130,120 @@ __optional__
+
+Once configured, you do not need to specify `creds`, `orgId`, or `keyspaces` in the WITH clause.
+
+[[inline-parameters]]
+=== Set Parameters Inline
+
+You can set the `creds`, `orgId`, and `keyspaces` options directly in the `WITH` clause of the USING AI statement. 
+If specified, these values override the `natural_cred`, `natural_orgid`, and `natural_context` parameters. 
+See <<example-3>>.
 
 == Result
 
 A JSON object containing the generated statement, execution status, and metrics.
 
 If the generated statement is a SELECT query and the `execute` option is `TRUE` (default), the output includes the query results as well.
 
+[[handling-passwords]]
 == Handling Passwords Securely
 
 Use `creds` and `natural_cred` parameters with caution to avoid exposing password information through history files or logs.
 When working with the cbq shell, avoid passing passwords directly on the command line. 
 Instead, use the `\set` command to specify only the username, and then enter the password at the terminal prompt.
 This ensures that the password is not recorded in the shell history.
 
-Similarly, if you are sending requests directly to a REST endpoint from a shell, be mindful of how you provide the password and whether it will be recorded in the shell history.
+For example: 
+
+[source, sh]
+----
+cbq> \set -natural_cred username@example.com;
+Enter password for "natural_cred": <enter password>
+
+cbq> \set;
+ Query Parameters : 
+ Parameter name : natural_cred
+ Value : [username@example.com:***]
+
+ Parameter name : profile
+ Value : ["timings"]
+
+ ...
+----
+
+Similarly, if you're sending requests directly to a REST endpoint from a shell, be mindful of how you provide the password and whether it will be recorded in the shell history.
 Consider using a method like the following to prompt for the password:
 
-[source,sh]
+[source,console]
 ----
 echo -n "Enter your password: "
 read -s p
 echo
 curl -s -d "natural_cred=<your-username>:${p}" ...
 ----
 
-If you choose to reuse the password by setting an evironment variable, note that it might be visible to other users on the system with sufficient privileges for process inspection (for example, through `/proc`).
+If you choose to reuse the password by setting an environment variable, note that it might be visible to other users on the system with sufficient privileges for process inspection (for example, through `/proc`).
 If security is a concern, consider using an HTTPS connection. 
 
 == Examples
 
-In the following examples, replace `natural_cred` and `natural_orgid` with your Capella credentials and organization ID, respectively.
+In the following examples, replace `natural_cred`/`creds` and `natural_orgid`/`orgId` with your Couchbase Capella credentials and organization ID, respectively.
 
 [[example-1]]
-.Natural language prompt with default request parameters
+.USING AI with default request parameters
 ====
 {example-note}
 
 .Request
-[source, console]
+[source, sh]
```

Comment:
This might seem counterintuitive, but I actually use `sqlpp` as the file type for cbq shell commands. Because they actually are (mainly) SQL++ queries with interspersed shell commands. See the [cbq](https://docs.couchbase.com/server/current/n1ql/n1ql-intro/cbq.html) documentation for consistency.


```suggestion
[source, sqlpp]
```